### PR TITLE
Support set test mode for flags test

### PR DESCRIPTION
**Pull request by hi-rustin**
See https://github.com/grafana/phlare/pull/497#issuecomment-1438260905

### CODE REVIEW

File: cmd/phlare/main.go [hunk 1]
```diff
@@ -49,14 +53,20 @@ func main() {
 		flags mainFlags
 	)
 
-	if err := cfg.DynamicUnmarshal(&flags, os.Args[1:], flag.CommandLine); err != nil {
+	if err := cfg.DynamicUnmarshal(&flags, os.Args[1:], flag.CommandLine, testMode); err != nil {
```
Comment (hi-rustin) [inline cmd/phlare/main.go:56]: I am not sure does this is a good way. Maybe it is too ugly. If you have any suggestions please feel free to comment. Thanks!

File: cmd/phlare/main.go [hunk 2]
```diff
@@ -16,6 +16,10 @@ import (
 	_ "github.com/grafana/phlare/pkg/util/build"
 )
 
+// testMode is used to test flag parsing.
+// We don't want to exit the process when running tests.
+var testMode = false
```
Comment (simonswine) [inline cmd/phlare/main.go:21]: I would move this as a private global to `pkg/cfg/dynamic.go`.

Then I would have a public method `SetTestMode(bool)` in `pkg/cfg/dynamic_test.go`. This doesn't pollute the arguments. How do you feel about that?

File: cmd/phlare/main.go (hunk 1 above)
Comment (simonswine) [reply cmd/phlare/main.go:56]: https://github.com/grafana/phlare/pull/542/files#r1116866328

File: cmd/phlare/main.go (hunk 2 above)
Comment (hi-rustin) [reply cmd/phlare/main.go:21]: Yeah. It sounds good! Thanks for your suggestion!

**Review by hi-rustin — COMMENTED** [review]
Self check

**Review by simonswine — APPROVED** [review]
Thanks for addressing this case :+1: 

LGTM

**Pull request merged**